pages/api/v1/delegate-chat/send-message.ts

```diff
@@ -0,0 +1,116 @@
+import messages from '~src/auth/utils/messages';
+import authServiceInstance from '~src/auth/auth';
+import storeApiKeyUsage from '~src/api-middlewares/storeApiKeyUsage';
+import { chatMessagesRef, chatDocRef } from '~src/api-utils/firestore_refs';
+import getSubstrateAddress from '~src/util/getSubstrateAddress';
+import { getDelegatesData } from '../delegations/getAllDelegates';
+
+const validateRequest = (req: NextApiRequest, network: string) => {
+	const { address, senderAddress, receiverAddress, content, chatId } = req.body;
+
+	if (!address || !senderAddress.length || !receiverAddress.length || !content.length || !chatId.length) {
+		return { error: messages.INVALID_PARAMS };
+	}
+
+	if (!(getEncodedAddress(String(address), network) || isAddress(String(address)) || getEncodedAddress(String(receiverAddress), network) || isAddress(String(receiverAddress)))) {
+		return { error: 'Invalid address' };
+	}
+
+	return { address, chatId, content, receiverAddress, senderAddress };
+};
+
+const validateParticipants = async (network: string, senderAddress: string, receiverAddress: string) => {
+	const senderSubstrateAddress = getSubstrateAddress(senderAddress);
+	const receiverSubstrateAddress = getSubstrateAddress(receiverAddress);
+
+	if (!senderSubstrateAddress || !receiverSubstrateAddress || senderAddress === receiverAddress) {
+		return { error: 'Invalid senderAddress or receiverAddress' };
+	}
+
+	const { data: delegatesList, error } = await getDelegatesData(network, receiverAddress);
+	if (error) return { error: error.message || messages.API_FETCH_ERROR };
+
+	if (delegatesList) {
+		const isValidReceiverAddress = delegatesList.some((delegate: IDelegateAddressDetails) => getSubstrateAddress(delegate.address) === receiverSubstrateAddress);
+		const isValidSenderAddress = delegatesList.some((delegate: IDelegateAddressDetails) => getSubstrateAddress(delegate.address) === senderSubstrateAddress);
+
+		if (!isValidReceiverAddress && !isValidSenderAddress) {
+			return { error: 'At least one participant must be a delegate' };
+		}
+	}
+
+	return { receiverSubstrateAddress, senderSubstrateAddress };
+};
+
+async function handler(req: NextApiRequest, res: NextApiResponse<IMessage | MessageType>) {
+	storeApiKeyUsage(req);
+
+	const network = String(req.headers['x-network']);
+	if (!network || !isValidNetwork(network)) return res.status(400).json({ message: 'Invalid network in request header' });
+
+	const token = getTokenFromReq(req);
+	if (!token) return res.status(400).json({ message: 'Invalid token' });
+
+	const user = await authServiceInstance.GetUser(token);
+	if (!user) return res.status(403).json({ message: messages.UNAUTHORISED });
+
+	const validationResult = validateRequest(req, network);
+	if ('error' in validationResult) return res.status(400).json({ message: validationResult?.error ?? messages.INVALID_PARAMS });
+
+	const { senderAddress, receiverAddress, content, chatId, senderImage, senderUsername } = req.body;
```

Comment:
There is no check for if any of logged in user's addresses is the same as recieved in the request body